apps/mobile/src/app/wallet/developer-console/bitcoin-scratch-pad.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState } from 'react';
+import { TextInput } from 'react-native';
+
+import { usePsbtSigner } from '@/features/psbt-signer/psbt-signer';
+import { useBitcoinAccounts } from '@/store/keychains/bitcoin/bitcoin-keychains.read';
+import { BitcoinLoader } from '@/store/keychains/keychains';
+import { t } from '@lingui/macro';
+import * as btc from '@scure/btc-signer';
+
+import { payerToBip32Derivation, payerToTapBip32Derivation } from '@leather.io/bitcoin';
+import { Box, Button, Text } from '@leather.io/ui/native';
+
+export default function BitcoinScratchPad() {
+  const { sign } = usePsbtSigner();
+  const [text, setText] = useState('bc1qetcn9fdtzq3wez6vrljjyry225ml7j52h3kwgl');
+
+  const accounts = useBitcoinAccounts().fromAccountIndex('d7d96884', 0);
+
+  const nativeSegwitAccount = accounts[0];
+  const taprootAccount = accounts[1];
+
+  if (!nativeSegwitAccount || !taprootAccount) throw new Error('No signer found');
+
+  const nativeSegwitPayer = nativeSegwitAccount.derivePayer({ receive: 0, addressIndex: 0 });
+  const taprootPayer = taprootAccount.derivePayer({ receive: 0, addressIndex: 0 });
+
+  return (
+    <Box p="5">
+      <BitcoinLoader fingerprint="d7d96884" accountIndex={0}>
+        {({ nativeSegwit, taproot }) => (
+          <Text fontSize={6} lineHeight={8}>
+            {JSON.stringify(
+              [nativeSegwit, taproot].map(({ keychain, ...rest }) => rest),
+              null,
+              2
+            )}
+          </Text>
+        )}
+      </BitcoinLoader>
+      <Text>{t`BitcoinScratchPad`}</Text>
+
+      <Text fontWeight="bold" mt="2">{t`Recipient`}</Text>
+      <TextInput
+        placeholder={t`Type here to translate!`}
+        onChangeText={newText => setText(newText)}
+        defaultValue={text}
+        autoCapitalize="none"
+      />
+
+      <Button
+        buttonState="default"
+        title={t`Sign`}
+        mt="5"
+        onPress={() => {
+          const tx = new btc.Transaction();
+          tx.addInput({
+            txid: '19c52a02cf783c73028b32aabad3b97afec7958052a45bd1504b84900e419bc6',
+            index: 0,
+            bip32Derivation: [payerToBip32Derivation(nativeSegwitPayer)],
```

Comment:
See here how, when creating a transaction, we decorate the transaction with `bip32Derivation` or `tapBip32Derivation`. This is required for Ledger transactions, and is also useful for our software implementation.

If we omit this, by the time the transaction reaches PsbtSigner, we won't know which key corresponds to each input. In our extension, we use an InputSigningConfig that includes metadata indicating which key owns each input. This works, but isn't ideal as we 1) have to create our own kinda schema to live alongside a PSBT, 2) duplicate the information anyway to support Ledger.

Much of the new logic this PR is to make it trivial to declare, and later read, the key derivation info of each key, such that by the time we get to signing a tx, we have full instructions on how to sign a input.